radicle/src/ssh.rs

```diff
@@ -75,30 +78,81 @@ pub mod fmt {
     }
 }
 
+#[derive(Debug, Error)]
+pub enum SignatureError {
+    #[error(transparent)]
+    Invalid(#[from] crypto::Error),
+    #[error(transparent)]
+    Encoding(#[from] encoding::Error),
+    #[error("unknown algorithm '{0}'")]
+    UnknownAlgorithm(String),
+}
+
+impl Encodable for crypto::Signature {
+    type Error = SignatureError;
+
+    fn read_ssh(r: &mut encoding::Cursor) -> Result<Self, Self::Error> {
+        let buf = r.read_string()?;
+        let mut r2 = buf.reader(0);
+
+        let sig_type = r2.read_string()?;
+        if sig_type != b"ssh-ed25519" {
+            return Err(SignatureError::UnknownAlgorithm(
+                String::from_utf8_lossy(sig_type).to_string(),
+            ));
+        }
+        let sig = crypto::Signature::try_from(r2.read_string()?)?;
+
+        Ok(sig)
+    }
+
+    fn write_ssh<E: Encoding>(&self, buf: &mut E) {
+        let mut buf2 = Vec::new();
+        buf2.extend_ssh_string(b"ssh-ed25519");
+        buf2.extend_ssh_string(self.as_ref());
+        buf.extend_ssh_string(&buf2);
```

Comment:
The numbered bufs are confusing, maybe you can give one a name, eg. `buf2` could be called `inner` or `wrapped` or something.